toolkits/slack/arcade_slack/tools/chat.py

```diff
@@ -72,43 +91,532 @@ def send_dm_to_user(
+                conversation_id = conversation["id"]
+                break
+
+        if not conversation_next_cursor:
             break
 
-    if not channel_id:
+    if not conversation_id:
         raise RetryableToolError(
-            "Channel not found",
-            developer_message=f"Channel with name '{channel_name}' not found.",
-            additional_prompt_content=format_channels(channels_response),
-            retry_after_ms=500,  # Play nice with Slack API rate limits
+            "Conversation not found",
+            developer_message=f"Conversation with name '{conversation_name}' not found.",
+            additional_prompt_content=f"Available conversation names: {conversation_names}",
+            retry_after_ms=500,
+        )
+
+    return await get_members_from_conversation_by_id(
+        context=context,
+        conversation_id=conversation_id,
+        limit=limit,
+        next_cursor=next_cursor,
+    )
+
+
+# TODO: make the function accept a current unix timestamp argument to allow testing without
+# mocking. Have to wait until arcade.core.annotations.Inferrable is implemented, so that we
+# can avoid exposing this arg to the LLM.
+@tool(
+    requires_auth=Slack(
+        scopes=["channels:history", "groups:history", "im:history", "mpim:history"],
+    )
+)
+async def get_conversation_history_by_id(
+    context: ToolContext,
+    conversation_id: Annotated[str, "The ID of the conversation to get history for"],
+    oldest_relative: Annotated[
+        Optional[str],
+        (
+            "The oldest message to include in the results, specified as a time offset from the "
+            "current time in the format 'DD:HH:MM'"
+        ),
+    ] = None,
+    latest_relative: Annotated[
+        Optional[str],
+        (
+            "The latest message to include in the results, specified as a time offset from the "
+            "current time in the format 'DD:HH:MM'"
+        ),
+    ] = None,
+    oldest_datetime: Annotated[
+        Optional[str],
+        (
+            "The oldest message to include in the results, specified as a datetime object in the "
+            "format 'YYYY-MM-DD HH:MM:SS'"
+        ),
+    ] = None,
+    latest_datetime: Annotated[
+        Optional[str],
+        (
+            "The latest message to include in the results, specified as a datetime object in the "
+            "format 'YYYY-MM-DD HH:MM:SS'"
+        ),
+    ] = None,
+    limit: Annotated[
+        Optional[int], "The maximum number of messages to return. Defaults to 20."
+    ] = MAX_PAGINATION_LIMIT,
+    cursor: Annotated[Optional[str], "The cursor to use for pagination. Defaults to None."] = None,
+) -> Annotated[
+    dict,
+    (
+        "The conversation history / messages and next cursor for paginating results (when "
+        "there are additional messages to retrieve)."
+    ),
+]:
+    """Get the history of a conversation in Slack."""
+    error_message = None
+    if oldest_datetime and oldest_relative:
```

Comment:
is this datetime handling useful anywhere else? in the future? 

if so, let's extract into a function